test/e2e/app/app.go

```diff
@@ -452,22 +444,21 @@ func (app *Application) ExtendVote(_ context.Context, req *abci.RequestExtendVot
 // without doing anything about them. In this case, it just makes sure that the
 // vote extension is a well-formed integer value.
 func (app *Application) VerifyVoteExtension(_ context.Context, req *abci.RequestVerifyVoteExtension) (*abci.ResponseVerifyVoteExtension, error) {
-	// We allow vote extensions to be optional
+	appHeight, areExtensionsEnabled := app.checkHeightAndExtensions(false, req.Height, "VerifyVoteExtension")
+	if !areExtensionsEnabled {
+		panic(fmt.Errorf("received call to VerifyVoteExtension at height %d, when vote extensions are disabled", appHeight))
+	}
+	// We don't allow vote extensions to be optional
 	if len(req.VoteExtension) == 0 {
+		app.logger.Error("received empty vote extension")
 		return &abci.ResponseVerifyVoteExtension{
-			Status: abci.ResponseVerifyVoteExtension_ACCEPT,
+			Status: abci.ResponseVerifyVoteExtension_REJECT,
```

Comment:
Is it not allowed to have an empty vote extension?

Reply:
Not in the current application's logic. I chose to do it this way (more stringent) to be able to better detect bugs... and it worked ;-)